File: 프로그래머스/2/49993. 스킬트리/스킬트리.py
from collections import deque

def solution(skill, skill_trees):
    answer = 0
    
    for skill_tree in skill_trees:
        skill_tree = deque(skill_tree)
        
        skill_level = 0
        flag = True
        for skill_set in skill_tree:
            if skill_set in skill:
                if skill[skill_level] == skill_set:
                    skill_level += 1
                else:
                    flag = False
                    break
        if flag:
            answer += 1
    
    return answer

File: 프로그래머스/2/49993. 스킬트리/스킬트리.js
function solution(skill, skill_trees) {
    let answer = 0;
    
    const convertSkillTrees = skill_trees.map((skills) => skills.split("").filter((item) => skill.includes(item)).join(""));
    convertSkillTrees.forEach((skillTree) => {
        if (skillTree === skill.slice(0, skillTree.length)) answer++;
    })
    
    return answer;
}